components/trustpilot/common/api-client.mjs

```diff
@@ -0,0 +1,136 @@
+import { axios } from "@pipedream/platform";
+import {
+  BASE_URL, HTTP_STATUS, RETRY_CONFIG,
+} from "./constants.mjs";
+import {
+  formatQueryParams, sleep,
+} from "./utils.mjs";
+
+/**
+ * Make an authenticated request to the Trustpilot API
+ * @param {object} trustpilotApp - The Trustpilot app instance with auth credentials
+ * @param {object} options - Request options
+ * @param {string} options.endpoint - API endpoint path
+ * @param {string} [options.method="GET"] - HTTP method
+ * @param {object} [options.params={}] - Query parameters
+ * @param {object} [options.data] - Request body data
+ * @param {object} [options.additionalHeaders={}] - Additional headers to include in the request
+ * @param {number} [options.timeout=30000] - Request timeout
+ * @param {number} [retries=RETRY_CONFIG.MAX_RETRIES] - Number of retries for rate limiting
+ * @returns {Promise<object>} API response data
+ */
+export async function makeRequest(trustpilotApp, {
+  endpoint,
+  method = "GET",
+  params = {},
+  data = null,
+  timeout = 30000,
+  additionalHeaders = {},
+  ...args
+}, retries = RETRY_CONFIG.MAX_RETRIES) {
+  const url = `${BASE_URL}${endpoint}`;
+  const headers = {
+    ...getAuthHeaders(trustpilotApp, url),
+    ...additionalHeaders,
+  };
+
+  const config = {
+    method,
+    url,
+    headers,
+    params: formatQueryParams(params),
+    timeout,
+    ...args,
+  };
+
+  if (data) {
+    config.data = data;
+  }
+
+  try {
+    const response = await axios(trustpilotApp, config);
+    return response.data || response;
+  } catch (error) {
+    if (retries > 0 && error.response?.status === HTTP_STATUS.TOO_MANY_REQUESTS) {
```

Comment:
I'm almost sure the error thrown by `@pipedream/platform`'s axios has the "status" object directly in it, not in a "response" key - this needs to be tested